public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -54,60 +54,62 @@ export function createSpanLinkFactory({
 
   let scopedVars = scopedVarsFromTrace(trace);
   const hasLinks = dataFrame.fields.some((f) => Boolean(f.config.links?.length));
-  const legacyFormat = dataFrame.fields.length === 1;
-
-  if (legacyFormat || !hasLinks) {
-    // if the dataframe contains just a single blob of data (legacy format) or does not have any links configured,
-    // let's try to use the old legacy path.
-    // TODO: This was mainly a backward compatibility thing but at this point can probably be removed.
-    return legacyCreateSpanLinkFactory(
-      splitOpenFn,
-      // We need this to make the types happy but for this branch of code it does not matter which field we supply.
-      dataFrame.fields[0],
-      traceToLogsOptions,
-      traceToMetricsOptions,
-      createFocusSpanLink,
-      scopedVars
-    );
-  }
 
-  if (hasLinks) {
-    return function SpanLink(span: TraceSpan): SpanLinks | undefined {
+  const createSpanLinks = legacyCreateSpanLinkFactory(
+    splitOpenFn,
+    // We need this to make the types happy but for this branch of code it does not matter which field we supply.
+    dataFrame.fields[0],
+    traceToLogsOptions,
+    traceToMetricsOptions,
+    createFocusSpanLink,
+    scopedVars
+  );
+
+  return function SpanLink(span: TraceSpan): SpanLinks | undefined {
+    let spanLinks = createSpanLinks(span);
```

Comment:
bit unsure about this change. Why do we now always use the legacy factory? I thought we could be able to just delete it by now.

Reply:
@aocenas - The legacy factory adds all loki, splunk, elastic/opensearch, and falcon logscale links. I could see correlations replacing this at some point, but I wouldn't want to remove that functionality now. Correlations will be released behind a feature flag.

Reply:
ah ok, hmm I am wondering if this wasn't a bug in a way. If there were links in the dataFrame itself then it didn't add the configured links for logs and metrics I assume. cc @joey-grafana btw this is also something we should change. I think we could just create the links in the data source and add them into dataFrame and handle them as any other links on the dataFrame. We would lose the categories I guess but as I mention here https://github.com/grafana/grafana/pull/67160#discussion_r1187394230 maybe we should lose it anyway. In general, feel bad that something that was legacy and I meant to delete it, ended up accumulating so much more functionality :D

Reply:
@aocenas I also could not be understanding that code. I didn't really test how existing links work with traces, I just looked at what the code does, but maybe it is never ran. I'll look into this.

Reply:
I'd say we should keep the types, but yeah simplifying the logic here and removing the `const legacyFormat = dataFrame.fields.length === 1;` check so that both data source config links and links in `dataframe.config.links` will be added to the menu is a good move.